src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -1983,10 +2083,30 @@ instance Semigroup LimitClause where
 
 instance Monoid LimitClause where
   mempty = Limit mzero mzero
-  mappend = (<>)
 
 -- | A locking clause.
-type LockingClause = Monoid.Last LockingKind
+data LockingClause =
+  GeneralLockingClause LockingKind
+  -- ^ Locking clause not specific to any database implementation
+  | PostgresLockingClause PostgresLockingKind
+  -- ^ Locking clause specific to postgres
+  | NoLockingClause
+
+-- | Monoid instance prioritization is:
+-- 1. "Strongest" lock (FOR UPDATE > FOR SHARE) if comparing two postgres
+-- specific locks of differing strength
+-- 2. Rightmost (last) locking clause otherwise
```

Comment:
🤔 

This behavior feels weird. Are the `LockingKind` constructors all more strict than the `PostgresLockingClause`? If I wrote `locking $ forUpdateOfSkipLocked myTable` and composed it with a query that had something more permissive, it would be weird for it to be something more permissive, only because it used a generic constructor instead of a Postgres constructor.

Reply:
Hmm the reason I did it this way is because I didn't want to change the existing behavior of the general locking clause - I could potentially prioritize the more "specific" instance and then within general locking kinds take the rightmost and within postgres take the strongest?